hoomd/hpmc/ComputeSDF.h

```diff
@@ -364,13 +390,184 @@ template<class Shape> void ComputeSDF<Shape>::countHistogram(uint64_t timestep)
+                                // expensive than the other
+                                //
+                                // first check for any hard overlaps
+                                // if there is one for a given scale value, there is no need to
+                                // check for any soft overlaps from m_mc.m_patch
+                                bool hard_overlap = detail::test_scaled_overlap<Shape>(
+                                    r_ij,
+                                    quat<Scalar>(orientation_i),
+                                    quat<Scalar>(orientation_j),
+                                    params[__scalar_as_int(postype_i.w)],
+                                    params[__scalar_as_int(postype_j.w)],
+                                    scale_factor);
+                                if (hard_overlap)
+                                    {
+                                    // add appropriate weight to appropriate bin and exit the
+                                    // loop over bins
+                                    hist_weight_ptl_i = 1.0; // = 1-e^(-\infty)
+                                    min_bin_ptl_i = bin_to_sample;
+                                    break;
+                                    } // end if overlap
+
+                                // if no hard overlap, check for a soft overlap if we have
+                                // patches
+                                if (m_mc->m_patch)
+                                    {
+                                    // compute the energy at this size of the perturbation and
+                                    // compare to the energy in the unperturbed state
+                                    vec3<Scalar> r_ij_scaled = r_ij * (Scalar(1.0) - scale_factor);
+                                    double u_ij_new
+                                        = m_mc->m_patch->energy(r_ij_scaled,
+                                                                typ_i,
+                                                                quat<float>(shape_i.orientation),
+                                                                float(h_diameter.data[i]),
+                                                                float(h_charge.data[i]),
+                                                                typ_j,
+                                                                quat<float>(orientation_j),
+                                                                float(h_diameter.data[j]),
+                                                                float(h_charge.data[j]));
+                                    // if enery has changed, there is a new soft overlap
+                                    // add the appropriate weight to the appropriate bin of the
+                                    // histogram and break out of the loop over bins
+                                    if (u_ij_new != u_ij_0)
```

Comment:
Is this the condition we want or would we want some small numerical tolerance here, such as 1e-14 or such.

Reply:
This method is valid for patch energies with step functions. So yes, this is the right condition.